lib/plugins/amplitudePlugin.ts

```diff
@@ -0,0 +1,120 @@
+import { add, init, track } from '@amplitude/analytics-node'
+import type {
+  AmplitudeReturn,
+  BaseEvent,
+  NodeOptions,
+  Plugin,
+  Result,
+} from '@amplitude/analytics-types'
+import type {
+  FastifyInstance,
+  FastifyReply,
+  FastifyRequest,
+  HookHandlerDoneFunction,
+} from 'fastify'
+import fp from 'fastify-plugin'
+
+async function plugin(
+  fastify: FastifyInstance,
+  config: AmplitudeConfig,
+  next: (err?: Error) => void,
+) {
+  if (!config.isEnabled) {
+    return next()
+  }
+
+  if (!config.apiKey) {
+    throw Error('Amplitude key not defined')
```

Comment:
shouldn't next with an error be invoked here?

Reply:
You are completely right! Actually, that approach looks better than the `throw` one